internal/client/api_account.go

```diff
@@ -0,0 +1,188 @@
+		return ResponseAccount{}, fmt.Errorf("creating account %s failed. %w: %s", acc.Name, ErrUnmarshaling, err)
+	}
+
+	return resAcc, nil
+}
+
+// DeleteAccount deletes a user from in Enzi.
+func (c *Client) ApiDeleteAccount(ctx context.Context, id string) error {
+	url := fmt.Sprintf("%s/%s", URLTargetForAccounts, id)
+	req, err := c.RequestFromTargetAndBytesBody(ctx, http.MethodDelete, url, []byte{})
+
+	if err != nil {
+		return fmt.Errorf("deleting account %s failed. %w: %s", id, ErrRequestCreation, err)
+	}
+
+	if _, err = c.doAuthorizedRequest(req); err != nil {
+		return fmt.Errorf("deleting account %s failed. %w", id, err)
+	}
+	return nil
+}
+
+// ReadAccount method retrieves a user from the enzi endpoint.
+func (c *Client) ApiReadAccount(ctx context.Context, id string) (ResponseAccount, error) {
+	url := fmt.Sprintf("%s/%s", URLTargetForAccounts, id)
+	req, err := c.RequestFromTargetAndBytesBody(ctx, http.MethodGet, url, []byte{})
+	if err != nil {
+		return ResponseAccount{}, fmt.Errorf("reading account %s failed. %w: %s", id, ErrRequestCreation, err)
+	}
+
+	body, err := c.doAuthorizedRequest(req)
+	if err != nil {
+		return ResponseAccount{}, fmt.Errorf("reading account %s failed. %w", id, err)
+	}
+
+	resAcc := ResponseAccount{}
+	if err := body.JSONMarshallBody(&resAcc); err != nil {
+		return ResponseAccount{}, fmt.Errorf("reading account %s failed. %w: %s", id, ErrUnmarshaling, err)
+	}
+	return resAcc, nil
+}
+
+// UpdateAccount updates a user in the enzi endpoint.
+func (c *Client) ApiUpdateAccount(ctx context.Context, id string, acc UpdateAccount) (ResponseAccount, error) {
+	url := fmt.Sprintf("%s/%s", URLTargetForAccounts, id)
+
+	req, err := c.RequestFromTargetAndJSONBody(ctx, http.MethodPatch, url, acc)
+
+	if err != nil {
+		return ResponseAccount{}, fmt.Errorf("updating account %s failed. %w: %s", id, ErrRequestCreation, err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	body, err := c.doAuthorizedRequest(req)
+	if err != nil {
+		return ResponseAccount{}, fmt.Errorf("updating account %s failed. %w", id, err)
+	}
+
+	resAcc := ResponseAccount{}
+	if err := body.JSONMarshallBody(&resAcc); err != nil {
+		return ResponseAccount{}, fmt.Errorf("updating account %s failed. %w: %s", id, ErrUnmarshaling, err)
+	}
+	return resAcc, nil
+}
+
+// ReadAccounts method retrieves all accounts depending on the filter passed from the enzi endpoint.
+func (c *Client) ApiReadAccounts(ctx context.Context, accFilter AccountFilter) ([]ResponseAccount, error) {
+	// req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.createEnziUrl("accounts"), nil)
+	req, err := c.RequestFromTargetAndBytesBody(ctx, http.MethodGet, URLTargetForAccounts, []byte{})
+	if err != nil {
+		return []ResponseAccount{}, fmt.Errorf("reading accounts in bulk '%s' failed. %w: %s",
+			accFilter.APIFormOfFilter(), ErrRequestCreation, err)
+	}
+
+	q := req.URL.Query()
+	q.Add("filter", accFilter.APIFormOfFilter())
+	req.URL.RawQuery = q.Encode()
```

Comment:
I feel that it might be worth while to build a function for this query arg management. It is used like this a couple of times. If you agree, I can write it and refactor outside of the scope of this ticket.